src/rtc/core.py

```diff
@@ -152,6 +153,304 @@ def save_as_cog(filename, scratch_dir = '.', logger = None,
+    tile_x_array[1], tile_y_array[1], z = transformation.TransformPoint(
+        tile_max_x_utm, tile_max_y_utm, elevation)
+    tile_x_array[2], tile_y_array[2], z = transformation.TransformPoint(
+        tile_max_x_utm, tile_min_y_utm, elevation)
+    tile_x_array[3], tile_y_array[3], z = transformation.TransformPoint(
+        tile_min_x_utm, tile_min_y_utm, elevation)
+    tile_min_y = np.min(tile_y_array)
+    tile_max_y = np.max(tile_y_array)
+    tile_min_x = np.min(tile_x_array)
+    tile_max_x = np.max(tile_x_array)
+
+    # handles antimeridian: tile_max_x around +180 and tile_min_x around -180
+    # add 360 to tile_min_x, so it becomes a little greater than +180
+    if tile_max_x > tile_min_x + 340:
+        tile_min_x, tile_max_x = tile_max_x, tile_min_x + 360
+
+    tile_ring = ogr.Geometry(ogr.wkbLinearRing)
+    tile_ring.AddPoint(tile_min_x, tile_max_y)
+    tile_ring.AddPoint(tile_max_x, tile_max_y)
+    tile_ring.AddPoint(tile_max_x, tile_min_y)
+    tile_ring.AddPoint(tile_min_x, tile_min_y)
+    tile_ring.AddPoint(tile_min_x, tile_max_y)
+    tile_polygon = ogr.Geometry(ogr.wkbPolygon)
+    tile_polygon.AddGeometry(tile_ring)
+    tile_polygon.AssignSpatialReference(polygon_srs)
+    return tile_polygon, tile_min_y, tile_max_y, tile_min_x, tile_max_x
+
+
+
+def _antimeridian_crossing_requires_special_handling(
+        file_srs, file_min_x, tile_min_x, tile_max_x):
+    '''
+    Check if ancillary input requires special handling due to
+    the antimeridian crossing
+
+    Parameters
+    ----------
+    file_srs: osr.SpatialReference
+        Ancillary file spatial reference system (SRS)
+    file_min_x: float
+        Ancillary file min longitude value in degrees
+    tile_min_x: float
+        MGRS tile min longitude value in degrees
+    tile_max_x: float
+        MGRS tile max longitude value in degrees
+
+    Returns
+    -------
+    flag_requires_special_handling : bool
+        Flag that indicate if the ancillary input requires special handling
+    '''
+
+    # Flag to indicate if the if the MGRS tile crosses the antimeridian.
+    flag_tile_crosses_antimeridian = tile_min_x < 180 and tile_max_x >= 180
+
+    # Flag to test if the ancillary input file is in geographic
+    # coordinates and if its longitude domain is represented
+    # within the [-180, +180] range, rather than the [0, +360] interval.
+    # This is verified by the test `min_x < -170`. There's no specific reason
+    # why -170 is used. It could be -160, or even 0.
+    flag_input_geographic_and_longitude_not_0_360 = \
+        file_srs.IsGeographic() and file_min_x < -170
+
+    # If both are true, tile requires special handling due to the
+    # antimeridian crossing
+    flag_requires_special_handling = (
+        flag_tile_crosses_antimeridian and
+        flag_input_geographic_and_longitude_not_0_360)
+
+    return flag_requires_special_handling
+
+
+
+def check_ancillary_inputs(check_ancillary_inputs_coverage,
```

Comment:
I guess this is basically copied from PROTEUS. I would suggest to modify the terminology accordingly to RTC.
In specific, if I understand this correctly, `tile` should be the geogrid of output.